internal/cmd/providers.go

```diff
@@ -35,24 +36,72 @@ func newProvidersCmd(cli *CLI) *cobra.Command {
 	return cmd
 }
 
+type providerAPIOptions struct {
+	PrivateKey  string
+	ClientEmail string
+	DomainAdmin string
+}
+
+func (o providerAPIOptions) Validate(providerKind string) error {
+	if providerKind != "google" {
```

Comment:
This doesn't match the comment below?

Reply:
It does a check if any Google specific fields are specified (which aren't applicable since this isn't a Google provider) and errors if they are. Maybe the negative wording is a bit confusing?